.github/workflows/ci.yml

```diff
@@ -2,58 +2,112 @@ name: CI
 on:
   - push
   - pull_request
+
 jobs:
-  build_and_test:
-    name: many-framework build and tests
+  build:
+    name: ${{ matrix.make.name }} (${{ matrix.os }})
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-22.04]
+        os: [ubuntu-latest, macos-latest]
+        rust: [nightly]
+        make:
+          - name: Clippy
+            task: "check-clippy"
+          - name: Unit tests
+            task: "build-all-test run-all-unit-test"
+          - name: Doc tests
+            task: "run-all-doc-test"
+          - name: Coverage
+            task: "coverage/report.lcov"
+        include:
+          - os: ubuntu-latest
+            sccache-path: /home/runner/.cache/sccache
+          - os: macos-latest
+            sccache-path: /Users/runner/Library/Caches/Mozilla.sccache
+        exclude:
+          - os: macos-latest
+            rust: nightly
+            make:
+              name: Clippy
+          - os: macos-latest
+            rust: nightly
+            make:
+              name: Coverage
+
+    env:
+      RUST_BACKTRACE: full
+      RUSTC_WORKSPACE_WRAPPER: sccache
```

Comment:
We might want to disable incremental compilation. 

See https://github.com/mozilla/sccache#rust

> Incrementally compiled crates cannot be cached. By default, in the debug profile Cargo will use incremental compilation for workspace members and path dependencies. [You can disable incremental compilation.](https://doc.rust-lang.org/cargo/reference/profiles.html#incremental)